flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStore.java

```diff
@@ -138,7 +138,6 @@ public void createDirtyResultInternal(JobResultEntry jobResultEntry) throws IOEx
         final Path path = constructDirtyPath(jobResultEntry.getJobId());
         try (OutputStream os = fileSystem.create(path, FileSystem.WriteMode.NO_OVERWRITE)) {
             mapper.writeValue(os, new JsonJobResultEntry(jobResultEntry));
```

Comment:
Hmm, I'm still not sure about this double close thingy 🤔 Can we simply make suure the `writeValue` doesn't close anything instead?

```suggestion
            mapper.writeValue(new NonClosingOutpusStreamDecorator(os), new JsonJobResultEntry(jobResultEntry));
```

nit: there is a typo in `NonClosingOutpusStreamDecorator` class name